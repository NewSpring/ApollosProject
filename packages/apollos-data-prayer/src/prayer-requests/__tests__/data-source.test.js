import ApollosConfig from '@apollosproject/config';
import PrayerRequest from '../data-source';

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://beta-rock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://beta-rock.newspring.cc/GetImage.ashx',
  },
});

// Builds a mock data source for Auth
const buildDataSource = (Auth) => {
  const dataSource = new PrayerRequest();
  dataSource.context = { dataSources: { Auth } };
  return dataSource;
};

// Fake user data
const AuthMock = { getCurrentPerson: () => ({ primaryAliasId: 123 }) };

describe('PrayerRequest', () => {
  it('constructs', () => {
    expect(new PrayerRequest()).toBeTruthy();
  });

  it('gets all public prayer requests', () => {
    const dataSource = new PrayerRequest();
    dataSource.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
          firstName: 'Isaac',
          lastName: 'Hardy',
          text: 'Pray this works.',
          createdByPersonAliasId: 447217,
        },
        {
          id: 'PrayerRequest:57c465ee3cd69524d729569b338607de',
          firstName: 'Rich',
          lastName: 'Dubee',
          text: 'Help me',
          createdByPersonAliasId: 447217,
        },
      ])
    );
    const result = dataSource.getAll();
    expect(result).resolves.toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
  });

  it('gets prayer request by id', () => {
    const dataSource = new PrayerRequest();
    dataSource.get = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
      })
    );
    // "1" is a specific prayer request id
    const result = dataSource.getFromId(1);
    expect(result).resolves.toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
  });

  it('gets all public prayer requests by campus', () => {
    const dataSource = new PrayerRequest();
    dataSource.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
          firstName: 'Isaac',
          lastName: 'Hardy',
          text: 'Pray this works.',
          createdByPersonAliasId: 447217,
        },
      ])
    );

    // "16" is a campus id
    const result = dataSource.getAllByCampus(16);
    expect(result).resolves.toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
  });

  it('gets current persons prayer requests', () => {
    const dataSource = buildDataSource(AuthMock);
    dataSource.get = jest.fn(() =>
      Promise.resolve([
        {
          id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
          firstName: 'Isaac',
          lastName: 'Hardy',
          text: 'Pray this works.',
          createdByPersonAliasId: 447217,
        },
      ])
    );

    const result = dataSource.getFromCurrentPerson();
    expect(result).resolves.toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
  });

  it('creates prayer request', async () => {
    const dataSource = buildDataSource(AuthMock);

    dataSource.post = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
      })
    );

    dataSource.get = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
      })
    );

    const result = await dataSource.add({
      FirstName: 'Test',
      LastName: 'bro',
      Text: 'Pray this works.',
      CategoryId: 2,
      CampusId: 16,
      IsPublic: true,
    });
    delete dataSource.post.mock.calls[0][1].EnteredDateTime;
    expect(result).toMatchSnapshot();
    expect(dataSource.post.mock.calls).toMatchSnapshot();
  });

  it('increments prayed for a request', async () => {
    const dataSource = buildDataSource(AuthMock);

    dataSource.put = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
        prayerCount: 8,
      })
    );

    dataSource.get = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
        prayerCount: 8,
      })
    );

    const result = await dataSource.incrementPrayed(10);
    expect(result).toMatchSnapshot();
    expect(dataSource.put.mock.calls).toMatchSnapshot();
  });

  it('flags a prayer request', async () => {
    const dataSource = buildDataSource(AuthMock);

    dataSource.put = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
        prayerCount: 8,
      })
    );

    dataSource.get = jest.fn(() =>
      Promise.resolve({
        id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
        firstName: 'Isaac',
        lastName: 'Hardy',
        text: 'Pray this works.',
        createdByPersonAliasId: 447217,
        prayerCount: 8,
      })
    );

    const result = await dataSource.flag(10);
    expect(result).toMatchSnapshot();
    expect(dataSource.put.mock.calls).toMatchSnapshot();
  });
});
