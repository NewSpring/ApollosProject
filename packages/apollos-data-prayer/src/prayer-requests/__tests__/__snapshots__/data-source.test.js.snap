// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrayerRequest creates prayer request 1`] = `
Object {
  "createdByPersonAliasId": 447217,
  "firstName": "Isaac",
  "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
  "lastName": "Hardy",
  "text": "Pray this works.",
}
`;

exports[`PrayerRequest creates prayer request 2`] = `
Array [
  Array [
    "/PrayerRequests",
    Object {
      "CampusId": 16,
      "CategoryId": 2,
      "FirstName": "Test",
      "IsActive": true,
      "IsApproved": true,
      "IsPublic": true,
      "LastName": "bro",
      "RequestedByPersonAliasId": 123,
      "Text": "Pray this works.",
    },
  ],
]
`;

exports[`PrayerRequest gets all public prayer requests 1`] = `
Array [
  Array [
    "PrayerRequests/Public",
    Object {},
    Object {},
  ],
]
`;

exports[`PrayerRequest gets all public prayer requests 2`] = `
Array [
  Object {
    "createdByPersonAliasId": 447217,
    "firstName": "Isaac",
    "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
    "lastName": "Hardy",
    "text": "Pray this works.",
  },
  Object {
    "createdByPersonAliasId": 447217,
    "firstName": "Rich",
    "id": "PrayerRequest:57c465ee3cd69524d729569b338607de",
    "lastName": "Dubee",
    "text": "Help me",
  },
]
`;

exports[`PrayerRequest gets all public prayer requests by campus 1`] = `
Array [
  Array [
    "PrayerRequests/Public?%24filter=CampusId%20eq%2016",
    Object {},
    Object {},
  ],
]
`;

exports[`PrayerRequest gets all public prayer requests by campus 2`] = `
Array [
  Object {
    "createdByPersonAliasId": 447217,
    "firstName": "Isaac",
    "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
    "lastName": "Hardy",
    "text": "Pray this works.",
  },
]
`;

exports[`PrayerRequest gets current persons prayer requests 1`] = `Array []`;

exports[`PrayerRequest gets current persons prayer requests 2`] = `
Array [
  Object {
    "createdByPersonAliasId": 447217,
    "firstName": "Isaac",
    "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
    "lastName": "Hardy",
    "text": "Pray this works.",
  },
]
`;

exports[`PrayerRequest gets prayer request by id 1`] = `
Array [
  Array [
    "PrayerRequests/1",
    Object {},
    Object {},
  ],
]
`;

exports[`PrayerRequest gets prayer request by id 2`] = `
Object {
  "createdByPersonAliasId": 447217,
  "firstName": "Isaac",
  "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
  "lastName": "Hardy",
  "text": "Pray this works.",
}
`;

exports[`PrayerRequest increments prayed for a request 1`] = `
Object {
  "createdByPersonAliasId": 447217,
  "firstName": "Isaac",
  "id": "PrayerRequest:b36e55d803443431e96bb4b5068147ec",
  "lastName": "Hardy",
  "prayerCount": 8,
  "text": "Pray this works.",
}
`;

exports[`PrayerRequest increments prayed for a request 2`] = `
Array [
  Array [
    "PrayerRequests/Prayed/10",
    Object {},
  ],
]
`;
